internal/server/server.go

```diff
@@ -579,6 +595,29 @@ func (s *Server) VerifyIdentityHeader(nextHandler echo.HandlerFunc) echo.Handler
 	}
 }
 
+func (s *Server) ValidateRequest(nextHandler echo.HandlerFunc) echo.HandlerFunc {
+	return func(ctx echo.Context) error {
+		request := ctx.Request()
+		loader := openapi3.NewSwaggerLoader()
+
+		route, params, err := s.router.FindRoute(request)
+		if err != nil {
+			return echo.NewHTTPError(http.StatusBadRequest, err)
```

Comment:
I think here it would be an internal server error, no? Since any non-existing route would have returned 404 before it ever hits this. So just `return err`.